login.js

```diff
@@ -0,0 +1,124 @@
+
+function isPasswordValid(password) {
+    return password.length >= 8;
+}
+
+function isUser(email, password) {
+    return USER_DATA.some((user) => user.email === email && user.password === password);
+}
+
+function validForm(email, password) {
+    return isEmailValid(email) && isPasswordValid(password);
+}
+
+function emailCheck() {
+    const email = checkEmail.value;
+    checkEmail.addEventListener('focusout', () => {
+        if (!isEmailValid(email)) {
+            checkEmail.classList.add('error');
+            errMessage.style.display = 'block';
+            //checkEmail.focus();
+        } else {
+            checkEmail.classList.remove('error');
+            errMessage.style.display = 'none';
+            loginBtn.classList.toggle('disabled', !validForm(email, checkPW.value));
+        }
+    });
+}
+
+function passwordCheck() {
+    const password = checkPW.value;
+    checkPW.addEventListener('focusout', () => {
+        if (isEmailValid(email) && !isPasswordValid(password)) {
+            checkPW.classList.add('error');
+            PerrMessage.style.display = 'block';
+            checkPW.focus();
+        } else {
+            checkPW.classList.remove('error');
+            PerrMessage.style.display = 'none';
+            loginBtn.classList.toggle('disabled', !validForm(checkEmail.value, password));
+        }
+    });
+}
+
+// function isValid() {
+//     const email = checkEmail.value;
+//     const password = checkPW.value;
+
+//     checkEmail.addEventListener('focusout', () => {
+//         if (!isEmailValid(email)) {
+//             checkEmail.classList.add('error');
+//             errMessage.style.display = 'block';
+//             checkEmail.focus();
+//         } else {
+//             checkEmail.classList.remove('error');
+//             errMessage.style.display = 'none';
+//         }
+//     });
+//     checkPW.addEventListener('focusout', () => {
+//         if (isEmailValid(email) && !isPasswordValid(password)) {
+//             checkPW.classList.add('error');
+//             PerrMessage.style.display = 'block';
+//             checkPW.focus();
+//         } else {
+//             checkPW.classList.remove('error');
+//             PerrMessage.style.display = 'none';
+//         }
+//     });
+//     if (validForm(email, password)) {
+//         loginBtn.classList.remove('disabled');
+//     } else {
+//         loginBtn.classList.add('disabled');
+//     }
+// }
+
+// checkEmail.addEventListener('focusout', isValid);
+checkEmail.addEventListener('input', emailCheck);
+// checkPW.addEventListener('focusout', isValid);
+checkPW.addEventListener('input', passwordCheck);
+
+loginBtn.addEventListener('click', (e) => {
+    e.preventDefault();
+    const email = checkEmail.value;
+    const password = checkPW.value;
+    if (!loginBtn.classList.contains('disabled')) {
+        if (isUser(email, password)) {
+            window.location.href = 'items.html';
+        } else {
+            showModal();
+        }
+    }
```

Comment:
@shinji530 여기는 아래처럼 코드를 간단하게 할 수 있어요!
코드 양이 많아질수록 scope의 depth가 깊으면 유지보수가 어려워져서 가급적이면 depth를 신경쓰는게 좋아요! 
```suggestion
    if (loginBtn.classList.contains('disabled')) return
    if (!isUser(email, password)) return showModal();
    window.location.href = 'items.html';
```

Reply:
넵 참고하겠습니다!!